dataclients/kubernetes/definitions/routegroupvalidator.go

```diff
@@ -0,0 +1,208 @@
+package definitions
+
+import (
+	"github.com/zalando/skipper/eskip"
+)
+
+type RouteGroupValidator struct{}
+
+var defaultRouteGroupValidator = &RouteGroupValidator{}
+
+// ValidateRouteGroup validates a RouteGroupItem
+func ValidateRouteGroup(rg *RouteGroupItem) error {
+	return defaultRouteGroupValidator.Validate(rg)
+}
+
+func ValidateRouteGroups(rgl *RouteGroupList) error {
+	var err []error
+	for _, rg := range rgl.Items {
+		// TODO: Use errors.Join
+		err = append(err, defaultRouteGroupValidator.Validate(rg))
+	}
+	return errorsJoin(err...)
+}
+
+func (rgv *RouteGroupValidator) Validate(item *RouteGroupItem) error {
+	e := rgv.basicValidation(item)
+	if e != nil {
+		return e
+	}
+	var err []error
+	err = append(err, rgv.filtersValidation(item))
+	err = append(err, rgv.predicatesValidation(item))
+
+	return errorsJoin(err...)
+}
+
+// TODO: we need to pass namespace/name in all errors
+func (rgv *RouteGroupValidator) basicValidation(r *RouteGroupItem) error {
+	// has metadata and name:
+	if r == nil || validate(r.Metadata) != nil {
+		return errRouteGroupWithoutName
+	}
+
+	// has spec:
+	if r.Spec == nil {
+		return routeGroupError(r.Metadata, errRouteGroupWithoutSpec)
+	}
+
+	if err := r.Spec.validate(); err != nil {
+		return routeGroupError(r.Metadata, err)
+	}
+
+	return nil
+}
+
+func (rgv *RouteGroupValidator) filtersValidation(item *RouteGroupItem) error {
+	var err []error
+	for _, r := range item.Spec.Routes {
+		for _, f := range r.Filters {
+			_, e := eskip.ParseFilters(f)
+			// TODO: Use errors.Join
+			err = append(err, e)
+		}
+	}
+
+	return errorsJoin(err...)
+}
+
+func (rgv *RouteGroupValidator) predicatesValidation(item *RouteGroupItem) error {
+	var err []error
+	for _, r := range item.Spec.Routes {
+		for _, p := range r.Predicates {
+			_, e := eskip.ParsePredicates(p)
```

Comment:
Here we actually allow a list of predicates and I think dataclient does the same while parsing.
This is an undocumented feature that might be surprising for users.
Lets address this in a separate PR after we check existing cluster resources.

Reply:
Same for the filters.